x-pack/plugins/lists/server/routes/import_list_item_route.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter } from 'kibana/server';
+
+import { LIST_ITEM_URL } from '../../common/constants';
+import {
+  buildRouteValidation,
+  buildSiemResponse,
+  transformError,
+  validate,
+} from '../siem_server_deps';
+import {
+  ImportListItemSchema,
+  acknowledgeSchema,
+  importListItemQuerySchema,
+  importListItemSchema,
+} from '../../common/schemas';
+
+import { getListClient } from '.';
+
+export const importListItemRoute = (router: IRouter): void => {
+  router.post(
+    {
+      options: {
+        body: {
+          output: 'stream',
+        },
+        tags: ['access:lists'],
+      },
+      path: `${LIST_ITEM_URL}/_import`,
+      validate: {
+        body: buildRouteValidation<typeof importListItemSchema, ImportListItemSchema>(
+          importListItemSchema
+        ),
+        query: buildRouteValidation(importListItemQuerySchema),
+      },
+    },
+    async (context, request, response) => {
+      const siemResponse = buildSiemResponse(response);
+      try {
+        const { list_id: listId, type } = request.query;
+        const lists = getListClient(context);
+        if (listId != null) {
+          const list = await lists.getList({ id: listId });
+          if (list == null) {
+            return siemResponse.error({
+              body: `list id: "${listId}" does not exist`,
+              statusCode: 409,
+            });
+          }
+          await lists.importListItemsToStream({
+            listId,
+            meta: undefined,
+            stream: request.body.file,
+            type: list.type,
+          });
+
+          return response.accepted({
+            body: {
+              acknowledged: true,
+            },
+          });
+        } else if (type != null) {
+          const { filename } = request.body.file.hapi;
+          // TODO: Should we prevent the same file from being uploaded multiple times?
```

Comment:
Could we surface up more info on imports? Even on a successful import I wasn't seeing any sort of response message showing up. And maybe on duplicates (looks like right now we're not allowing and just returning the existing list?) also add a message letting user know of current behavior?

Reply:
Duplicates are tricky as its an async operation when it does imports at a particular lower level. The counting of the duplicates and what not become wrong values or false values as the streams can be independent parallel things.

I noticed a lot of odd streaming things and I don't want to add penalties of waiting in any other areas as I think we want to introduce streaming of large list items as a feature from the REST endpoint all the way to the backend and hopefully 🤞back-pressure works auto-magically with the streams from the browser to the backend. 

If the imports become pure streaming in, we might be able to re-introduce duplicate counting but the way this works right now it "sinks" all the lists into memory on the import and then begins streaming that list into the backend and returns the 200 so it does not time out.

Reply:
I will bubble up more information here on an import where it auto-creates the list. I think that is a good idea.